update_projects_table_test.py

```diff
@@ -1,119 +1,132 @@
+"""Unit tests for the "update_projects_table" module."""
+
 from pathlib import Path
-import tempfile
-import unittest
 
+import pytest
 import update_projects_table
 
 
-class TestCreateProjectTable(unittest.TestCase):
-    """Unit tests for the "update_projects_table" module."""
+class TestExtractMetadata:
+    """Unit tests for the "extract_metadata" function."""
 
-    def generic_test_extract_metadata(self, input, expected):
-        actual = {}
-        with tempfile.NamedTemporaryFile(delete_on_close=False) as f:
-            if input:
-                f.write(input)
-            f.close()
-            actual = update_projects_table.extract_metadata(Path(f.name))
-        self.assertEqual(expected, actual)
+    @pytest.fixture
+    def tmp_file(self, tmp_path):
```

Comment:
I don't understand why do we need this fixture.

Reply:
It's a Thin wrapper over `tmp_path` to create and delete a temporary file inside it. Not strictly required because `tmp_path` does its own cleanup, but I still think it's nicer.